contracts/ETHRegistrarController.sol

```diff
@@ -175,4 +220,20 @@ contract ETHRegistrarController is Ownable {
 
         return cost;
     }
+
+    function _doRenew(string memory name, uint duration) internal returns(uint cost) {
+        uint cost = rentPrice(name, duration);
+        bytes32 label = keccak256(bytes(name));
+        uint expires = base.renew(uint256(label), duration);
+        emit NameRenewed(name, label, cost, expires);
+        return cost;
+    }
+    
+    function _sendReferralFee(address payable referrer, uint cost) internal {
```

Comment:
Do we not emit an event to track which referrer has received referral?

Reply:
Good point. I'll add something.